draft-ietf-anima-constrained-join-proxy.md

```diff
@@ -310,48 +308,77 @@ JPY[H(),C()] = Join Proxy message with header H and content C
-
 ~~~
     JPY_message =
     [
-       ip        : bstr,
-       port      : int,
-       family    : int,
-       ifindex   : int
+       pledge_context   : bstr,
        content   : bstr
     ]
 
 ~~~
 {: #fig-cddl title='CDDL representation of JPY message' align="left"}
 
-The contents are DTLS encrypted. In CBOR diagnostic notation the payload JPY[H(IP_P:p_P)], will look like:
+The Join Proxy cannot decrypt the DTLS payload and has no knowledge of the transported media type.
+The contents are DTLS encrypted.
 
-~~~
-      [h'IP_p', p_P, family, ident, h'DTLS-payload']
-~~~
+The context payload is to be reflected by the Registrar when sending reply packets to the Join Proxy.
+The context payload is not standardized.
+It is to be used by the Join Proxy to record which pledge the traffic came from.
 
-On reception by the Registrar, the Registrar MUST verify that the number of array elements is larger than or equal to 5, and reject the message when the number of array elements is smaller than 5.
-After replacing the 5th "content" element with the DTLS payload of the response message and leaving all other array elements unchanged, the Registrar returns the response message.
+The Join Proxy SHOULD encrypt this context with a symmetric key known only to the Join Proxy.
+This key need not persist on a long term basis, and MAY be changed periodically.
+The considerations of  {{Section 5.2 of RFC8974}} apply.
 
-Examples are shown in {{examples}}.
+This is intended to be identical to the mechanism described in {{Section 7.1 of RFC9031}}.
+However, since the CoAP layer is inside of the DTLS layer (which is between the Pledge and the Registrar), it is not possible for the Join Proxy to act as a CoAP proxy.
+
+A typical context parameter might be constructed with the following CDDL structure.
+(This is illustrative only: the contents are not subject to standardization)
 
-The header field is completely opaque to the receiver. A Registrar MUST copy the header and return it unmodified in the return message.
+~~~~
+    pledge_context_message = [
+      family:  uint .bits 1,
+      ifindex: uint .bits 8,
+      srcport: uint .bits 16,
+      iid:     bstr .bits 64,
+    ]
+~~~~
+
+This results in a total of 96 bits, or 12 bytes.
+This fits nicely into a single AES128 CBC block for instance, resulting in a 16 byte context message.
+The Join Proxy MUST maintain the same context block for all communications from the same pledge.
+This implies that any encryption key either does not change during the communication, or that when it does, it is acceptable to break any onboarding connections which have not yet completed.
+If using a construct like above, this should not be a problem to do without maintaining state.
+
+Note: only the lower 64-bits of the origin IP need to be recorded, because they are all IPv6 Link-Local addresses, so the upper 64-bits are just "fe80::"
```

Comment:
```suggestion
Note: when IPv6 is used only the lower 64-bits of the origin IP need to be recorded, because they are all IPv6 Link-Local addresses, so the upper 64-bits are just "fe80::". For IPv4, a Link-Local IPv4 address {{?RFC3927}} would be used, and it would fit into 64-bits.
```